packages/@aws-cdk/aws-appconfig-alpha/README.md

```diff
@@ -18,37 +18,88 @@
+const env = new appconfig.Environment(this, 'MyEnv', {
+  application: app,
+});
+
+new appconfig.HostedConfiguration(this, 'MyHostedConfig', {
+  application: app,
+  deployTo: [env],
+  content: appconfig.ConfigurationContent.fromInlineText('This is my configuration content.'),
+});
+
+```
+
+This will create the application and environment for your configuration and then deploy your configuration to the 
+specified environment.
+
+For more information about what these resources are: [Creating feature flags and free form configuration data in AWS AppConfig](https://docs.aws.amazon.com/appconfig/latest/userguide/creating-feature-flags-and-configuration-data.html).
+
+For more information about deploying configuration: [Deploying feature flags and configuration data in AWS AppConfig](https://docs.aws.amazon.com/appconfig/latest/userguide/deploying-feature-flags.html)
+
+____
+
+For an in-depth walkthrough of specific resources and how to use them, please take a look at the following sections.
 
 ## Application
 
-In AWS AppConfig, an application is simply an organizational construct like a folder. This organizational construct has a 
-relationship with some unit of executable code. For example, you could create an application called MyMobileApp to organize and 
-manage configuration data for a mobile application installed by your users. Configurations and environments are associated with 
-the application.
+[AWS AppConfig Application Documentation](https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-creating-namespace.html)
 
-The name and description of an application are optional.
+In AWS AppConfig, an application is simply an organizational 
+construct like a folder. Configurations and environments are 
+associated with the application.
+
+When creating an application through CDK, the name and 
+description of an application are optional.
 
 Create a simple application:
 
 ```ts
 new appconfig.Application(this, 'MyApplication');
 ```
 
-Create an application with a name and description:
+## Environment
+
+[AWS AppConfig Environment Documentation](https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-creating-environment.html)
+
+Basic environment with monitors:
 
 ```ts
-new appconfig.Application(this, 'MyApplication', {
-  applicationName: 'App1',
-  description: 'This is my application created through CDK.',
+declare const application: appconfig.Application;
+declare const alarm: cloudwatch.Alarm;
+declare const compositeAlarm: cloudwatch.CompositeAlarm;
+
+new appconfig.Environment(this, 'MyEnvironment', {
+  application,
+  monitors: [
+    appconfig.Monitor.fromCloudWatchAlarm(alarm),
+    appconfig.Monitor.fromCloudWatchAlarm(compositeAlarm),
+  ],
 });
 ```
 
+Environment monitors also support L1 CfnEnvironment.MonitorsProperty constructs. However, this is not the recommended approach 
+for CloudWatch alarms because a role will not be auto-generated if not provided.
```

Comment:
```suggestion
Environment monitors also support L1 `CfnEnvironment.MonitorsProperty` constructs. However, this is not the recommended approach 
for CloudWatch alarms because a role will not be auto-generated if not provided.
```

I actually have a few issues with this section:

- the sentence should mention _how_, i.e. by using the `fromCfnMonitorsProperty` method.
- however, I am not a fan of this approach at all. am I missing some context about why this method was added in the first place? CDK doesn't typically encourage this type of L1 method on L2 so I want to dig a bit deeper on why its necessary at all
- finally, looking at the code, it seems like the issue with `fromCfnMonitorsProperty` is that it will fail entirely if a role is not provided, likely in a confusing manner (because errors due to `!` don't tend to be very descriptive). Looking at this code below:

```ts
  public static fromCfnMonitorsProperty(monitorsProperty: CfnEnvironment.MonitorsProperty): Monitor {
    return {
      alarmArn: monitorsProperty.alarmArn!,
      alarmRoleArn: monitorsProperty.alarmRoleArn,
      monitorType: MonitorType.CFN_MONITORS_PROPERTY,
    };
  }
```

I think, if we have a reason to keep this method, it should be updated to (to avoid `!` entirely):

```ts
  public static fromCfnMonitorsProperty(monitorsProperty: CfnEnvironment.MonitorsProperty): Monitor {
    if (monitorsProperty.alarmArn === undefined) { throw new Error('need this prop'); }
    return {
      alarmArn: monitorsProperty.alarmArn,
      alarmRoleArn: monitorsProperty.alarmRoleArn,
      monitorType: MonitorType.CFN_MONITORS_PROPERTY,
    };
  }
```

Reply:
updated with suggested changes